timeseries/src/autogluon/timeseries/utils/timer.py

```diff
@@ -14,57 +48,101 @@ def __init__(
         self.start_time = None
 
     def start(self) -> Self:
+        """Start or reset the timer."""
         self.start_time = time.monotonic()
         return self
 
     def time_elapsed(self) -> float:
+        """Time since the timer was started. This method can also be used when
+        ``time_limit`` is set to None to count time forward (i.e., as opposed to
+        a countdown timer which other methods imply)."""
         if self.start_time is None:
             raise RuntimeError("Timer has not been started")
         return time.monotonic() - self.start_time
 
     def time_remaining(self) -> Optional[float]:
+        """Time remaining on the timer. If ``time_limit`` is None, this method also
+        returns None."""
         if self.start_time is None:
             raise RuntimeError("Timer has not been started")
         if self.time_limit is None:
             return None
         return self.time_limit - (time.monotonic() - self.start_time)
 
     def timed_out(self) -> bool:
+        """Whether the timer has timed out. If ``time_limit`` is None, this method
+        always returns False."""
+        if self.start_time is None:
+            raise RuntimeError("Timer has not been started")
         if self.time_limit is None:
             return False
         return self.time_elapsed() >= self.time_limit
 
 
 class SplitTimer(Timer):
```

Comment:
I am not sure if inheritance makes sense here given that the `SplitTimer` is only used via `get()` and `split()` methods, and the meaning of `timed_out()` and `time_remaining()` methods is not clear for the `SplitTimer` ("time out / time remaining per split or in total?").